### PR TITLE
Update security overview document for recent features

### PR DESCRIPTION
**Pull request by agross-oss**
This set of changes brings the security overview document up to date with the past 2 releases.

### CODE REVIEW

File: doc/security/security-overview.rst [hunk 1]
```diff
@@ -150,8 +150,17 @@ The security architecture is based on a monolithic design where the
...
 binary. System calls are implemented as function calls without requiring
 context switches. Static linking eliminates the potential for
-dynamically loading malicious code. Memory protection and task
-separation techniques are in scope starting with version 1.10.0.
+dynamically loading malicious code.
+
+Additional protection features are available in later releases.  Stack
+protection mechanisms are provided to protect against stack overruns.
+In addition, applications can take advantage of thread separation
+features to split the system into privileged and privileged execution
```
Comment (mbolivar) [inline doc/security/security-overview.rst]: privileged and unprivileged

File: doc/security/security-overview.rst [hunk 2]
```diff
@@ -150,8 +150,17 @@ The security architecture is based on a monolithic design where the
...
+environments.  Memory protection features provides the capability to
```
Comment (mbolivar) [inline doc/security/security-overview.rst]: provide

File: doc/security/security-overview.rst [hunk 3]
```diff
@@ -201,8 +210,8 @@ more detail; this document is part of that process.
 Execution Protection
 ====================
 
-Execution protection is planned for future releases and is roughly
-categorized into the following tasks:
+Execution protection is part of the project and can be categorized
```
Comment (mbolivar) [inline doc/security/security-overview.rst]: maybe just "supported"?

File: doc/security/security-overview.rst (hunk 1 above)
Comment (agross-linaro) [reply doc/security/security-overview.rst]: heh, my fingers didn't type what my mind was thinking.  good catch.

File: doc/security/security-overview.rst (hunk 2 above)
Comment (agross-linaro) [reply doc/security/security-overview.rst]: will fix

File: doc/security/security-overview.rst (hunk 3 above)
Comment (agross-linaro) [reply doc/security/security-overview.rst]: that simplifies the statement, will incorporate

**Review by mbolivar — APPROVED** [review]
LGTM

**Review by nashif — APPROVED** [review]

**Review by MaureenHelm — APPROVED** [review]

**Pull request merged**